apps/ntag/ntagCmdApp/ntag_main.cpp

```diff
@@ -0,0 +1,428 @@
+    blk[Ntag::NC_REG] |= 0x01;
+    if(!ntag.writeEeprom(0x39 * NTAG_BLOCK_SIZE, blk, NTAG_BLOCK_SIZE)) {
+        fprintf(stderr, "Failed in writing config register for NC_REG\n");
+	exit(1);
+    }
+*/
+
+    if(noEepromMem == true) {
+    	fprintf(stderr, "Eeprom hex of the tag is:\n");
+    	// Note readEeprom tacks on one 1 block to the block number - thus -1 on all block numbers
+    	for(i = 0 ; i < 0x7F ; i++) {
+	    //NT3H2111_2211 does not allow reads on these blocks
+	    if(i > 0x39 && i < 0x3F)
+	    	continue;
+	    fprintf(stderr, "Block %X:\t", 1 + i);
+    	    if(!ntag.readEeprom(i * NTAG_BLOCK_SIZE, blk, NTAG_BLOCK_SIZE)) {
+                fprintf(stderr, "Failed in reading block\n");
+	        exit(1);
+    	    }
+    	    if(le_hex_BinaryToString(blk, NTAG_BLOCK_SIZE, blkHex, sizeof(blkHex)) != -1) {
+                fprintf(stderr, "%s\n", blkHex);
+	        strcpy(ntagp, blkHex);
+	        ntagp += strlen(blkHex);
+	    }
+    	    else
+                fprintf(stderr, "Failed in call to le_hex_BinaryToString of Eeprom hex\n");
+    	}
+    }
+
+    dhubIO_PushString(TAG_MEM_EEPROM, DHUBIO_NOW, EepromHex);
+/*
+    // Let's read SRAM memory - for the NT3H2111_2211 blocks 0xF8 - 0xFB are sram
+    len = (0xFB - 0xF8 + 0x1) * NTAG_BLOCK_SIZE ;
+    if(!ntag.readSram(0, ntagp, len))
+    {
+        fprintf(stderr, "Failed in reading SRAM memory\n");
+	exit(2);
+    }
+
+    if(le_hex_BinaryToString(ntagp, len, EepromHex, sizeof(EepromHex)) != -1)
+        fprintf(stderr, "SRAM hex of the tag is: %s\n", EepromHex);
+    else
+        fprintf(stderr, "Failed in call to le_hex_BinaryToString\n");
+
+    ntagp += len;
+*/
+    dhubIO_PushString(TAG_MEM_SRAM, DHUBIO_NOW, "NOT IMPLEMENTED");
+
+    // Register reading is a byte based interface
+    for(i = 0 ; i <= 6 ; i++) {
+        if(!ntag.readRegister((Ntag::REGISTER_NR) i, reg)) {
+            fprintf(stderr, "Failed in reading Register: %d\n", i);
+	    exit(3);
+        }
+        else {
+	    //fprintf(stderr, "Session Register %d\t value: %X\n", i, reg);
+	    blk[i] = reg;
+        }
+    }
+
+    if(le_hex_BinaryToString(blk, 6, blkHex, sizeof(blkHex)) != -1)
+        fprintf(stderr, "Session Regs of the tag is: %s\n", blkHex);
+    else
+        fprintf(stderr, "Failed in call to le_hex_BinaryToString of Session Regs \n");
+
+    dhubIO_PushString(TAG_MEM_SESSION_REG, DHUBIO_NOW, blkHex);
+}
+
+//--------------------------------------------------------------------------------------------------
+COMPONENT_INIT
+{
+    // calling ntag without arguments will print the Usage
+    if (le_arg_NumArgs() < 1)
+    {
+        Usage();
+        exit(EXIT_SUCCESS);
+    }
+    else
+    {
+        const char *commandPtr = le_arg_GetArg(0);
+
+        if ((0 == strcmp(commandPtr, "help")) ||
+            (0 == strcmp(commandPtr, "--help")) ||
+            (0 == strcmp(commandPtr, "-h")))
+        {
+            Usage();
+            exit(EXIT_SUCCESS);
+        }
+        else
```

Comment:
same comment as above.  This is an `else` which follows an `if` statement that always exits